// ng-particles/__ivy_ngcc__/fesm2015/ng-particles.js
import { EventEmitter, Component, Input, Output, NgModule } from '@angular/core';
import { tsParticles } from 'tsparticles';
import * as ɵngcc0 from '@angular/core';
export * from 'tsparticles';

class NgParticlesComponent {
    constructor() {
        this.particlesLoaded = new EventEmitter();
    }
    ngAfterViewInit() {
        tsParticles.load(this.id, this.options).then(container => {
            this.particlesLoaded.emit(container);
        });
    }
}
NgParticlesComponent.ɵfac = function NgParticlesComponent_Factory(t) { return new (t || NgParticlesComponent)(); };
NgParticlesComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: NgParticlesComponent, selectors: [["Particles"]], inputs: { options: "options", id: "id" }, outputs: { particlesLoaded: "particlesLoaded" }, decls: 1, vars: 1, consts: [[3, "id"]], template: function NgParticlesComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("id", ctx.id);
    } }, encapsulation: 2 });
NgParticlesComponent.propDecorators = {
    options: [{ type: Input }],
    id: [{ type: Input }],
    particlesLoaded: [{ type: Output }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgParticlesComponent, [{
        type: Component,
        args: [{
                selector: 'Particles',
                template: `
      <div [id]="id"></div> `
            }]
    }], function () { return []; }, { particlesLoaded: [{
            type: Output
        }], options: [{
            type: Input
        }], id: [{
            type: Input
        }] }); })();

class NgParticlesModule {
}
NgParticlesModule.ɵfac = function NgParticlesModule_Factory(t) { return new (t || NgParticlesModule)(); };
NgParticlesModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgParticlesModule });
NgParticlesModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgParticlesModule, [{
        type: NgModule,
        args: [{
                declarations: [NgParticlesComponent],
                imports: [],
                exports: [NgParticlesComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgParticlesModule, { declarations: [NgParticlesComponent], exports: [NgParticlesComponent] }); })();

/*
 * Public API Surface of ng-particles
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgParticlesComponent, NgParticlesModule };
