// ng-particles/fesm2015/ng-particles.js
import { EventEmitter, Component, Input, Output, NgModule } from '@angular/core';
import { tsParticles } from 'tsparticles';
export * from 'tsparticles';

class NgParticlesComponent {
    constructor() {
        this.particlesLoaded = new EventEmitter();
    }
    ngAfterViewInit() {
        tsParticles.load(this.id, this.options).then(container => {
            this.particlesLoaded.emit(container);
        });
    }
}
NgParticlesComponent.decorators = [
    { type: Component, args: [{
                selector: 'Particles',
                template: `
      <div [id]="id"></div> `
            },] }
];
NgParticlesComponent.propDecorators = {
    options: [{ type: Input }],
    id: [{ type: Input }],
    particlesLoaded: [{ type: Output }]
};

class NgParticlesModule {
}
NgParticlesModule.decorators = [
    { type: NgModule, args: [{
                declarations: [NgParticlesComponent],
                imports: [],
                exports: [NgParticlesComponent]
            },] }
];

/*
 * Public API Surface of ng-particles
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgParticlesComponent, NgParticlesModule };
